src/connection.js

```diff
@@ -567,7 +570,8 @@ class Connection extends EventEmitter {
 
   clearRequestTimer() {
     if (this.requestTimer) {
-      return clearTimeout(this.requestTimer);
+      clearTimeout(this.requestTimer);
+      this.requestTimer = undefined;
```

Comment:
> It doesn't hurt to have a redundant call to clearRequestTimer().

Fair enough.

> Another solution would be to place it in the exit event of the SENT_CLIENT_REQUEST state.

That would be too late, isn't it? I think this only gets invoked after all the rows are consumed. You could add another redundant call on 'exit' to account for the case, if there is one, where neither 'data' nor 'message' is received.